syntax = "proto3";

package cloud;

service NodeService {
  rpc CreateNode(NodeRequest) returns (NodeResponse);
}

message NodeRequest {
  string user = 1;
  string node_type = 2; // e.g. "small", "medium", "gpu"
}

message NodeResponse {
  string result = 1;
  string node_id = 2;
}
